here-naksha-lib-psql/src/commonMain/kotlin/naksha/psql/executors/PgWriter.kt

```diff
@@ -226,10 +242,26 @@ class PgWriter(
                                 InsertFeature(session, writeExecutor).execute(collection, write)
                             )
                         } else {
-                            updateFeature(collection, previousMetadataProvider, write)
+                            val updatedTupleResult = UpdateFeature(
+                                session,
+                                previousMetadataProvider,
+                                writeExecutor
+                            ).execute(collection, write)
+                            if (updatedTupleResult.first == null) return ErrorResponse(updatedTupleResult.second,updatedTupleResult.third)
+                            updatePrevTupleCache(updatedTupleResult.first!!)
+                            cachedTupleNumber(write, updatedTupleResult.first!!)
                         }
 
-                    WriteOp.UPDATE -> updateFeature(collection, previousMetadataProvider, write)
+                    WriteOp.UPDATE -> {
+                        val updatedTupleResult = UpdateFeature(
+                            session,
+                            previousMetadataProvider,
+                            writeExecutor
+                        ).execute(collection, write)
+                        if (updatedTupleResult.first == null) return ErrorResponse(updatedTupleResult.second,updatedTupleResult.third)
+                        updatePrevTupleCache(updatedTupleResult.first!!)
+                        cachedTupleNumber(write, updatedTupleResult.first!!)
```

Comment:
Do not workaround the compiler, simply copy the value to stack:
```kotlin
val first = updatedTupleResult.first
if (first == null) return ErrorResponse(updatedTupleResult.second,updatedTupleResult.third)
updatePrevTupleCache(first)
cachedTupleNumber(write,first)
```
This not only shortens the code, it as well makes it better.